src/firebolt/async_db/cursor.py

```diff
@@ -164,24 +173,59 @@ def close(self) -> None:
         # remove typecheck skip  after connection is implemented
         self.connection._remove_cursor(self)  # type: ignore
 
-    def _store_query_data(self, response: Response) -> None:
+    def _append_query_data(self, response: Response) -> None:
         """Store information about executed query from httpx response."""
 
+        row_set: Tuple[
+            int, Optional[List[Column]], Optional[List[List[RawColType]]]
+        ] = (-1, None, None)
+
         # Empty response is returned for insert query
-        if response.headers.get("content-length", "") == "0":
-            return
-        try:
-            query_data = response.json()
-            self._rowcount = int(query_data["rows"])
-            self._descriptions = [
-                Column(d["name"], parse_type(d["type"]), None, None, None, None, None)
-                for d in query_data["meta"]
-            ]
-
-            # Parse data during fetch
-            self._rows = query_data["data"]
-        except (KeyError, JSONDecodeError) as err:
-            raise DataError(f"Invalid query data format: {str(err)}")
+        if response.headers.get("content-length", "") != "0":
+            try:
+                query_data = response.json()
+                rowcount = int(query_data["rows"])
+                descriptions = [
+                    Column(
+                        d["name"], parse_type(d["type"]), None, None, None, None, None
+                    )
+                    for d in query_data["meta"]
+                ]
+
+                # Parse data during fetch
+                rows = query_data["data"]
+                row_set = (rowcount, descriptions, rows)
+            except (KeyError, JSONDecodeError) as err:
+                raise DataError(f"Invalid query data format: {str(err)}")
+
+        self._row_sets.append(row_set)
+        if self._next_set_idx == 0:
+            # Populate values for first set
+            self._pop_next_set()
+
+    @check_not_closed
+    @check_query_executed
+    def nextset(self) -> Optional[bool]:
+        """
+        Skip to the next available set, discarding any remaining rows
+        from the current set.
+        Returns True if operation was successful,
+        None if there are no more sets to retrive
+        """
+        return self._pop_next_set()
+
+    def _pop_next_set(self) -> Optional[bool]:
+        """
+        Same functionality as .nextset, but doesn't check that query has been executed.
+        """
+        if self._next_set_idx >= len(self._row_sets):
+            return None
+        cur_set = self._row_sets[self._next_set_idx]
+        self._rowcount = cur_set[0]
```

Comment:
Can we use unpacking here: `self._rowcount, self._descriptions, self._rows = self._row_sets[self._next_set_idx]`?

Reply:
Yes, seems better